wgpu-core/src/instance.rs

```diff
@@ -894,8 +894,11 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
             // This means that backends which do provide accurate device types
             // will be preferred if their device type indicates an actual
             // hardware GPU (integrated or discrete).
-            PowerPreference::LowPower => integrated.or(discrete).or(other).or(virt).or(cpu),
-            PowerPreference::HighPerformance => discrete.or(integrated).or(other).or(virt).or(cpu),
+            Some(PowerPreference::LowPower) => integrated.or(discrete).or(other).or(virt).or(cpu),
+            Some(PowerPreference::HighPerformance) => {
+                discrete.or(integrated).or(other).or(virt).or(cpu)
+            }
+            None => None,
         };
 
         let mut selected = preferred_gpu.unwrap_or(0);
```

Comment:
I'm worried that this might cause us to make weird decisions like if the only vulkan adapter is software, picking it over a GL adapter.

Reply:
In order to work around this problem, we should find the lowest index adapter of (integrated, discrete, other) and choose that, otherwise choose zero.